aws_kms_pkcs11.cpp

```diff
@@ -401,20 +401,50 @@ CK_RV C_GetMechanismInfo(CK_SLOT_ID slotID, CK_MECHANISM_TYPE type, CK_MECHANISM
     if (pInfo == NULL) {
         return CKR_ARGUMENTS_BAD;
     }
+    AwsKmsSlot& slot = slots->at(slotID);
+    CK_ULONG keySize = 0;
     switch (type) {
     case CKM_RSA_PKCS:
-        pInfo->ulMinKeySize = 2048;
-        pInfo->ulMaxKeySize = 2048;
-        pInfo->flags = CKF_SIGN;
-        break;
+        switch(slot.GetKeySpec()) {
+        case Aws::KMS::Model::KeySpec::RSA_2048:
+            keySize = 2048;
+            break;
+        case Aws::KMS::Model::KeySpec::RSA_3072:
+            keySize = 3072;
+            break;
+        case Aws::KMS::Model::KeySpec::RSA_4096:
+            keySize = 4096;
+            break;
+        }
+
+        if (keySize > 0) {
+            break;
+        }
+        // invalid combination of mechanism and KMS key spec and
+        return CKR_MECHANISM_INVALID;
```

Comment:
Would work fine as it is, but as a matter of personal preference could you return this error as a default case in the above switch instead? (Then you can drop the if(keySize > 0) check and just break). Same with ECDSA below.